src/ir/from_ast/astconv.rs

```diff
@@ -223,17 +224,23 @@ impl<'ctx, 'prog> BuildCtx<'ctx, 'prog> {
 
     /// Forward declare an event without adding its delay. We need to do this
     /// since delays of events may mention the event itself.
-    fn declare_event(&mut self, eb: &ast::EventBind) -> EventIdx {
+    /// `interface_port` is the optional interface port associated with this event.
+    fn declare_event(
+        &mut self,
+        eb: &ast::EventBind,
+        interface_port: Option<ir::InfoIdx>,
+    ) -> EventIdx {
         let info = self.comp.add(ir::Info::event(
             eb.event.copy(),
             eb.event.pos(),
             eb.delay.pos(),
+            interface_port,
         ));
         // Add a fake delay of 0.
         let e = ir::Event {
             delay: self.comp.num(0).into(),
             info,
-            interface_port: None,
+            has_interface: true,
```

Comment:
It seems a bit weird to me that `has_interface` is unconditionally true? Shouldn't there some check to ensure that the interface port exists? If that happens at a later point, please add a comment pointing to the function

Reply:
Yep, this was a bug I found earlier yesterday and hadn't pushed yet, sorry.